### PR TITLE
ARD: Discovery protocol

### PR DESCRIPTION
**Pull request by oskarth**
Draft @dshulyak @adambabik possibly contains inaccuracies, please correct.

Mostly want something that other people with less context (chat/core/qa/pm) can use as a reference point.

### CODE REVIEW

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/fHZp5Z for branch PR-4245

<blockquote><div><strong><a href="https://i.diawi.com/fHZp5Z">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/KBEjaj for branch PR-4245

<blockquote><div><strong><a href="https://i.diawi.com/KBEjaj">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: doc/decisions/0005-discovery-protocol.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+- Use and enable Discovery v5 for all networks
+- Do this before beta is released
+
+This means we have capacity to achieve 5000+ DAU / 500 concurrent connection. It
+seems unlikely we can do so with reliable messaging in the current setup.
+
+Trade-off for this is:
+- Uncertainty in introducing a new protocol leading to possible delays
+- Possible increase in CPU/Network (we can disable protocol once peers are found though)
```
Comment (dshulyak) [inline doc/decisions/0005-discovery-protocol.md:33]: we do it already + we will hard stop discovery after certain timeout

**Review by dshulyak — APPROVED** [review]

**Review by adambabik — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

File: doc/decisions/0005-discovery-protocol.md (hunk 1 above)
Comment (oskarth) [reply doc/decisions/0005-discovery-protocol.md]: fixed

**Pull request merged**